src/io.rs

```diff
@@ -1,10 +1,12 @@
 //! Error types
```

Comment:
As per my summary, I belive we need to move all of this code to `esp_idf_svc::io::vfs`:
* If we are to switch to a simple global function - `set_stdio_mode` it can be exposed directly in the `vfs` module, without the need for a `stdio` sub-module.